src/main/java/com/sarality/db/InMemoryTable.java

```diff
@@ -41,8 +43,27 @@ public Long create(T data) {
 
   @Override
   public List<T> readAll(Query query) {
+
     List<T> dataList = new ArrayList<>();
-    dataList.addAll(dataMap.values());
+
+    // Null query, just return everything
+    if (query == null) {
+      dataList.addAll(dataMap.values());
+      return dataList;
+    }
+
+    // parse the query
+    QueryParser queryParser = new QueryParser(query);
+
+    // check if this is a supported query
+    if (!isSupportedQueryType(queryParser)) {
```

Comment:
nitpic: let's have a method for each type of supported query. In this case the method is

isPrimaryKeyQuery(parsedQuery)